test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -34,18 +39,24 @@ public MsbuildIntegrationTestFixture()
             var dotnetExecutableName = RuntimeEnvironmentHelper.IsWindows ? "dotnet.exe" : "dotnet";
             TestDotnetCli = Path.Combine(_cliDirectory, dotnetExecutableName);
 
-            var sdkPaths = Directory.GetDirectories(Path.Combine(_cliDirectory, "sdk"));
+            var sdkPath = Directory.GetDirectories(Path.Combine(_cliDirectory, "sdk")).Single();
 
+#if NETCOREAPP5_0
             // TODO - remove when shipping. See https://github.com/NuGet/Home/issues/8508
-            // const string dotnetMajorVersion = "3.";
-            const string dotnetMajorVersion = "5.";
-            PatchSDKWithCryptographyDlls(dotnetMajorVersion, sdkPaths);
+            PatchSDKWithCryptographyDlls(sdkPath);
+#endif
 
-            MsBuildSdksPath = Path.Combine(
-             sdkPaths.Where(path => path.Split(Path.DirectorySeparatorChar).Last().StartsWith(dotnetMajorVersion)).First()
-             , "Sdks");
+            MsBuildSdksPath = Path.Combine(sdkPath, "Sdks");
 
-            _templateDirectory = TestDirectory.Create();
+            _templateDirectory = new SimpleTestPathContext();
+            WriteGlobalJson(_templateDirectory.WorkingDirectory);
+            var addSourceArgs = new AddSourceArgs()
+            {
+                Configfile = _templateDirectory.NuGetConfig,
+                Name = "nuget.org",
+                Source = "https://api.nuget.org/v3/index.json"
```

Comment:
Why do we need this?

Reply:
Any test project that targets `netstandard2.0` gets an implicit package reference to `NETStandard.Library`, so that package and its dependencies are needed. However, the .NET 3.0 SDK and later no longer has a fallback folder, so these packages are no longer available locally.

So, either we need to have nuget.org as a package source, or we need to update all the test projects to use TFMs that do not need to download any packages. I'm concerned that even if we tartget `netstandard2.1` or `netcoreapp3.1`, the .NET 5 SDK might have an implicit `PackageDownload`, which would still need nuget.org.

Note that since your change to cache the templates in this folder, only the template folder needs the nuget.org package source. The test projects themselves use the template cache's global packages folder as a package source, so tests themselves can remain all-local.

Reply:
Do you think I should add this info as a comment in the code?

Reply:
added a comment.